### ISSUE TITLE
Coredump at launch with GLib-GIO error

### ISSUE THREAD

**Issue by adrinux** [onivim/oni2#3528]
Had a crash at launch - hadn't installed a new nightly since January so updated - still crashes at launch. Tried removing .config/oni2 for fresh settings, no extensions etc - still crashes.

Running with '-f --log-file=startup.log' doesn't write this error to the log but it appears in the terminal after:
```bash
(Oni2_editor:60844): GLib-GIO-ERROR **: 11:24:06.859: Settings schema 'org.gnome.settings-daemon.plugins.xsettings' does not contain a key named 'antialiasing'

Error: connect ECONNREFUSED /tmp/adrian/exthost-client-60844-c0a1d3-sock2 
    at PipeConnectWrap.afterConnect [as oncomplete] (net.js:1146:16) {
  errno: -111,
  code: 'ECONNREFUSED',
  syscall: 'connect',
  address: '/tmp/adrian/exthost-client-60844-c0a1d3-sock2'
}

```
journalctl has this kernel trap and systemd-coredump report:
```bash
May 11 11:24:06 zen audit[60844]: ANOM_ABEND auid=1000 uid=1000 gid=1000 ses=4 pid=60844 comm="Oni2_editor" exe="/tmp/adrian/.mount_OnivimTexcdd/usr/bin/Oni2_editor" sig=5 res=1
May 11 11:24:06 zen kernel: traps: Oni2_editor[60844] trap int3 ip:7ff87f2ddba8 sp:7ffdd03c0df0 error:0 in libglib-2.0.so.0.6800.1[7ff87f2a0000+8c000]
May 11 11:24:06 zen kernel: audit: type=1701 audit(1620728646.854:631): auid=1000 uid=1000 gid=1000 ses=4 pid=60844 comm="Oni2_editor" exe="/tmp/adrian/.mount_OnivimTexcdd/usr/bin/Oni2_editor" sig=5 res=1
-----
May 11 11:24:07 zen systemd-coredump[60880]: Process 60844 (Oni2_editor) of user 1000 dumped core.
                                             
                                             Stack trace of thread 60844:
                                             #0  0x00007ff87f2ddba8 g_log_structured_array (libglib-2.0.so.0 + 0x5aba8)
                                             #1  0x00007ff87f2dde76 g_log_default_handler (libglib-2.0.so.0 + 0x5ae76)
                                             #2  0x00007ff87f2df2b9 g_logv (libglib-2.0.so.0 + 0x5c2b9)
                                             #3  0x00007ff87f2df560 g_log (libglib-2.0.so.0 + 0x5c560)
                                             #4  0x00007ff87f4f0b72 n/a (libgio-2.0.so.0 + 0xdeb72)
                                             #5  0x00007ff87f4f237f g_settings_get_enum (libgio-2.0.so.0 + 0xe037f)
                                             #6  0x00007ff87fe7a99e n/a (n/a + 0x0)

```
This on Linux manjaro Gnome - stable...sort of...I briefly switched to unstable to pull in Gnome 40, so some packages may be from the 'unstable' manjaro Gnome branch.

Current versions:
**Gnome 40.0.0, using Wayland
glib2 2.68.1-1**

EDIT: to add
Presumably related to this [bug in GTK](https://gitlab.gnome.org/GNOME/gtk/-/issues/3680)
Current installed gtk3 package is: **1:3.24.29-1**

**Comment by bryphe** [comment]
Hmm, maybe related: https://gitlab.gnome.org/GNOME/gtk/-/issues/3680

**Comment by bryphe** [comment]
Reading through that thread - I'm not sure there's a fix on the Onivim side - it sounds like a GTK bug.

Does using one of these versions help?
- `gtk3-3.24.25-3.fc34.x86_64` (your version is newer, but it sounds like they backported to this version, and maybe didn't backport to 24.29)
- `gtk4-4.1.0-3.fc34.x86_64`

**Comment by adrinux** [comment]
Tried on a second machine - this one running manjaro Gnome unstable - exactly the same issue.

I was also thinking this was probably a Gtk bug but unlike the reports in that Gtk issue this __only__ affects Onivim.

The settings were [moved](https://gitlab.gnome.org/GNOME/gsettings-desktop-schemas/-/issues/31)

Somewhat reluctant to start reverting packages, will set up Timeshift for this second machine if I have time at the weekend and give that a shot. Might try compiling Onivim first though, to rule out something Appimage related.

**Comment by adrinux** [comment]
Built and installed the AppImage on the originally problematic desktop machine.
Everything appears to be working fine - so a workaround at least.

Doesn't really help figure out the original problem...let me know if there is anything else you want me to try, but at this point I think I'm happy for you to close the issue on the basis this is just some weird edge case during the transition to Gnome 40.

**Comment by oliveconstant** [comment]
Happening to me as well on Fedora 34 (GNOME/Wayland). Some info:

- oni2 0.5.9-nightly
- glibc.x86_64 2.33-14.fc34
- gtk3.x86_64 3.24.29-1.fc34
- gtk4.x86_64 4.2.1-1.fc34

**Comment by DominoTree** [comment]
I posted a quick fix (based on GNOME GitLab thread) in #3706
